README.adoc

```diff
@@ -263,11 +275,35 @@ run testscript.sh
 [ "$status" = "0" ]
 
 shellmock_verify
+[ "${capture[@]} = 2 ]
 [ "${capture[0]} = "grep-stub string1 file2" ]
 [ "${capture[1]} = "grep-stub string1 file3" ]
 
 ```
 
+===== Mock with whitespace in the parameters
+
+If the **grep** command is run by the script under test it will return a status 0 if arg1 is "string1" regardless of the rest of the args.  In order
+to verify that the function was called you would need to use **shellmock_verify** and do a comparison.
+
+If the --match argument were "'string1 string2' file", where the double quotes and single quotes are
+swapped, then shellmock treats the string as if it were '"string1 string2" file'.  That will be evident in the
+capture array.  This is due to how the shell treats " and '.  Shellmock can not always tell the difference so we have
+normalized them to double quotes.
+
+```bash
+shellmock_expect grep --status 0 --type partial --match '"string1 string2" file'
```

Comment:
Yes your understanding is correct.   That is because when the users script calls the mock the quotes are interpreted and eaten by the shell.
- grep "a b" bob.c
- grep 'a b' bob.c

The two above do the exact same thing and from inside the grep stub I can tell which was passed because there are no quotes at all any more.


As far as the documentation, that is a fair ask.  I was thinking also that the readme should maybe correlate with the samples.bats.  I could spend more time describing each in more detail.